src/run-on-golem/run-on-golem.action.ts

```diff
@@ -241,64 +252,62 @@ function installSignalHandlers(context: TaskAPIContext, events: EventEmitter<Tas
 
   // This is used to detect if the activity was terminated by the provider, error or timeout.
   // If it is, TaskExecutor is already shutting down. Make sure we terminate the shell.
-  events.on("golemEvents", async (e) => {
-    if (e instanceof Events.ActivityDestroyed) {
-      // This will happen on activity timeout
-      if (terminating || context.exited) return;
-      terminating = true;
-      console.log("Activity destroyed, terminating shell...");
-
-      await context.terminate();
-
-      process.exit(0);
-    }
-  });
-
-  events.on("taskFailed", async () => {
-    // This will happen on activity timeout and when executor times out waiting for offers
-    if (terminating || context.exited) return;
+  glm.activity.events.on("activityDestroyed", async () => {
+    // This will happen on activity timeout
+    if (terminating || context.metadata.terminating) return;
     terminating = true;
-    console.log("Terminating shell...");
+    console.log("Activity destroyed, terminating shell...");
 
-    await context.terminate();
+    await lease.finalize();
 
     process.exit(0);
   });
 }
 
 export async function runOnGolemAction(files: string[], options: RunOnGolemOptions) {
   // Prepare shell variables.
-  const vars: VarsType = {
+  const env: ProcessEnvVars = {
     ...(options.env ? process.env : {}),
   };
-
-  // Create task executor.
-  const executor = await createExecutor(options);
-  const context = new TaskAPIContext(executor, vars);
-  installSignalHandlers(context, executor.events);
+  const metadata = {
+    activityStart: new Date(),
+    terminating: false,
+  };
+  const context: ProcessContext = {
+    env,
+    metadata,
+  };
+  const glm = new GolemNetwork();
+  await glm.connect();
 
   try {
-    // Do magic so we have a work context.
-    await context.magic();
+    const order = await createMarketOrder(options);
+    const lease = await glm.oneOf(order);
+    // force deploy activity on the provider
```

Comment:
hmm, maybe the lease process should automatically initialize the activity. Now if someone gets a leaseProcess and doesn't call getExeUnit for a minute, the agreement will be terminated and the leaseProcess will be broken.. but that's for another task